assets/js/graphql/get-traits.js

```diff
@@ -0,0 +1,160 @@
+import { query } from './query.js';
+//import { modalLink } from './modal.js';
+
+
+/** The GraphQL query used to get traits. */
+export const getTraitsQuery = `
+query TraitQuery($pageSize: Int, $page: Int, $name: String, $studyType: String, $publicationId: String, $author: String) {
+    traits(pageSize: $pageSize, page: $page, name: $name, studyType: $studyType, publicationId: $publicationId, author: $author) {
+      results {
+        name
+        qtlStudy {
+          identifier
+          synopsis
+          description
+          genotypes
+          organism {
+            genus
+            species
+          }
+          dataSet {
+            publication {
+              firstAuthor
+              pubMedId
+              doi
+            }
+          }
+        }
+        gwas {
+          identifier
+          synopsis
+          description
+          genotypes
+          organism {
+            genus
+            species
+          }
+          dataSet {
+            publication {
+              firstAuthor
+              pubMedId
+              doi
+            }
+          }
+        }
+      }
+      pageInfo {
+        currentPage
+        pageSize
+        numResults
+        pageCount
+        hasPreviousPage
+        hasNextPage
+      }
+    }
+  }
+`;
+
+
+/**
+ * Gets traits from GraphQL.
+ * @param {object} queryData - An object containing zero or more variables for the GraphQL query.
+ * @param {object} pageData - An object containing pagination data for the GraphQL query, if any.
+ * @param {object} options - An object containing optional parameters for the HTTP request,
+ * namely, an optional `AbortSignal` instance that can be used to cancel the request mid-flight.
+ * @returns {Promise} A `Promise` that resolves to the result of the GraphQL query.
+ */
+export function getTraits(queryData={}, pageData={}, options={}) {
+  const genus = queryData.genus;
+  const species = queryData.species;
+  const studyType = queryData.type;
+  const name = queryData.traits;
+  const publicationId = queryData.pubId;
+  const author = queryData.author;
+  if (studyType === 'QTL') {
+    studyType = 'QTLStudy';
+  }
+  const {page, pageSize} = pageData;
+  const variables = {
+    genus,
+    species,
+    studyType,
+    name,
+    publicationId,
+    author,
+    page,
+    pageSize: 10,
+  };
+  const {abortSignal} = options;
+  return query(getTraitsQuery, variables, abortSignal);
+}
+
+/**
+ * Converts GraphQL `TraitsResults` into the `PaginatedSearchResults<TraitSearchResult[]>` used by the
```

Comment:
There is no "Trait Search" web component. All of the types mentioned in this comment should reference the "[Trait Association Search](https://github.com/legumeinfo/web-components/blob/main/src/lis-trait-association-search-element.ts)" web component's types.